core/src/main/java/com/graphhopper/GraphHopperReader.java

```diff
@@ -0,0 +1,138 @@
+        this.chHandler = chHandler;
+        this.locationIndex = locationIndex;
+    }
+
+    public static GraphHopperReader loadExisting(String graphCache, GraphConfig config, List<CHProfile> chProfiles) {
+        if (isEmpty(graphCache))
+            throw new IllegalStateException("GraphHopperLocation is not specified. Call setGraphHopperLocation or init before");
+
+        File tmpFileOrFolder = new File(graphCache);
+        if (!tmpFileOrFolder.isDirectory() && tmpFileOrFolder.exists())
+            throw new IllegalArgumentException("GraphHopperLocation cannot be an existing file. Has to be either non-existing or a folder.");
+
+        GHDirectory dir = new GHDirectory(graphCache, config.getDAType());
+        EncodingManager encodingManager = EncodingManager.create(new DefaultEncodedValueFactory(), new DefaultFlagEncoderFactory(), graphCache);
+        GraphHopperStorage graphHopperStorage = new GraphHopperStorage(dir, encodingManager, config.hasElevation(),
+                encodingManager.needsTurnCostsSupport(), config.getDefaultSegmentSize());
+        graphHopperStorage.addCHGraphs(chProfiles);
+        if (!graphHopperStorage.loadExisting())
+            throw new RuntimeException("Cannot load Graph from " + graphCache);
+
+        LocationIndexTree locationIndex = new LocationIndexTree(graphHopperStorage, dir);
+        if (!locationIndex.loadExisting())
+            throw new RuntimeException("Cannot load LocationIndex from " + graphCache);
+
+        CHPreparationHandler chHandler = CHPreparationHandler.loadExisting(graphHopperStorage, chProfiles);
+        return new GraphHopperReader(encodingManager, graphHopperStorage, chHandler, locationIndex);
+    }
+
+    // TODO NOW other customization mechanism than inheritance needed
+    public RoutingAlgorithmFactory getAlgorithmFactory(HintsMap map) {
+        RoutingAlgorithmFactory routingAlgorithmFactory = new RoutingAlgorithmFactorySimple();
+        if (chHandler.isEnabled())
+            return chHandler.getDecoratedAlgorithmFactory(routingAlgorithmFactory, map);
+
+        return routingAlgorithmFactory;
+    }
+
+    // TODO NOW other customization mechanism than inheritance needed
+    public Weighting createWeighting(HintsMap hintsMap, FlagEncoder encoder, TurnCostProvider turnCostProvider) {
+        String weightingStr = toLowerCase(hintsMap.getWeighting());
+        if ("shortest".equalsIgnoreCase(weightingStr)) {
+            return new ShortestWeighting(encoder, turnCostProvider);
+        } else if ("fastest".equalsIgnoreCase(weightingStr) || weightingStr.isEmpty()) {
+            if (encoder.supports(PriorityWeighting.class))
+                return new PriorityWeighting(encoder, hintsMap, turnCostProvider);
+
+            return new FastestWeighting(encoder, hintsMap, turnCostProvider);
+        }
+        throw new IllegalArgumentException("weighting " + weightingStr + " not supported");
+    }
+
+    public GHResponse route(GHRequest request) {
```

Comment:
Same here: I think routing and loading should not be in the same class (or if its in the same class it should be a very thin class that only delegates to other components that are either concerned with just the loading or just the routing)